synapse/handlers/oidc_handler.py

```diff
@@ -690,9 +691,17 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
                 self._render_error(request, "invalid_token", str(e))
                 return
 
+        # Pull out the user-agent and IP from the request.
+        user_agent = request.requestHeaders.getRawHeaders(b"User-Agent", default=[b""])[
+            0
+        ].decode("ascii", "surrogateescape")
+        ip_address = self.hs.get_ip_from_request(request)  # type: ignore
```

Comment:
I'm not sure why mypy didn't like this:
> synapse/handlers/oidc_handler.py:698: error: "HomeServer" has no attribute "get_ip_from_request"  [attr-defined]

It seems to not like the `HomeServer` object much...

Reply:
Probably because the `pyi` file hadn't declared it. Now that we have gotten rid of that it should no longer be a problem

Reply:
After rebasing this wasn't causing an issue anymore and I was able to remove it. Thanks for the explanation though! 👍